### PR TITLE
fix repay_borrow_all

### PR DESCRIPTION
**Pull request by GopherJ**

### CODE REVIEW

File: pallets/loans/src/lib.rs [hunk 1]
```diff
@@ -1015,10 +1016,10 @@ impl<T: Config> Pallet<T> {
             .checked_sub(repay_amount)
             .ok_or(ArithmeticError::Underflow)?;
         let total_borrows = Self::total_borrows(currency_id);
-        let total_borrows_new = total_borrows
-            .checked_sub(repay_amount)
-            .ok_or(ArithmeticError::Underflow)?;
+        // TODO : total_borrows should contain interests
+        let total_borrows_new = total_borrows.saturating_sub(repay_amount);
```
Comment (GopherJ) [inline pallets/loans/src/lib.rs:1020]: total_borrows < repay_amount and caused this issue:)

we need to find a way to make total_borrows contain also interests

Comment (GopherJ) [inline pallets/loans/src/lib.rs:1020]: total_borrows < repay_amount and caused this issue:)

we need to find a way to make total_borrows contain also interests

Comment (yz89) [reply pallets/loans/src/lib.rs:1020]: I think we can use this PR to fix #252. It only appears in the early days when total_borrows is too small.
When the case appears, the result of this modification is that `exchange_rate` won't be updated and no lender will get supply interest. That shouldn't be a problem.

Comment (yz89) [reply pallets/loans/src/lib.rs:1020]: Could you modify the TODO to a NOTE and explain the code briefly? And pls add an edge test to catch this case.

**Comment by yz89** [comment]
I think we can use this PR after addressed the comments.

File: pallets/loans/src/lib.rs (hunk 1 above)
Comment (GopherJ) [reply pallets/loans/src/lib.rs:1020]: ok I'll do it

File: pallets/loans/src/lib.rs [hunk 2]
```diff
@@ -1035,17 +1036,17 @@ impl<T: Config> Pallet<T> {
 
     // Calculates and returns the most recent amount of borrowed balance of `currency_id`
     // for `who`.
-    pub fn borrow_balance_stored(
+    pub fn recent_borrow_balance(
```
Comment (kaichaosun) [inline pallets/loans/src/lib.rs:1039]: The name refactor doesn't make sense, `recent` is too general.

Comment (GopherJ) [reply pallets/loans/src/lib.rs:1039]: do you have an suggestion here? like `latest`

Comment (yz89) [reply pallets/loans/src/lib.rs:1039]: Maybe `current`

**Comment by GopherJ** [comment]
@yz89  pls check latest commit, it'll catch this error if we change back to `checked_..`

**Review by yz89 — APPROVED** [review]

**Pull request merged**